test/unittest/CMakeLists.txt

```diff
@@ -23,6 +23,7 @@ add_subdirectory(dds/status)
 add_subdirectory(dds/subscriber)
 add_subdirectory(dds/topic)
 add_subdirectory(dds/topic/DDSSQLFilter)
+add_subdirectory(dds/xtypes/dyndata_idl)
```

Comment:
I think you got a bit confused with naming, I guess it should be `dyndata_json` instead.